### PR TITLE
Ensure busy-status in ui-multiplexer-particle during async processing

### PR DESCRIPTION
**Pull request by sjmiles**
Also add code to make slot-composer-test StorageNg compatible.

**note**: test still does not pass under StorageNg: no data flows into the multiplexer Collection.

### CODE REVIEW

File: src/tests/slot-composer-test.ts [hunk 1]
```diff
@@ -102,27 +102,22 @@ recipe
...
         .expectRenderSlot('C', 'otherSlot')
         ;
-
     await arc.instantiate(plan);
-    await arc.idle;
     await observer.expectationsCompleted();
   });
 
-  it('initialize recipe and render hosted slots', async () => {
+  it('initialize recipe and render hosted slots', async function() {
```
Comment (mmandlis) [inline src/tests/slot-composer-test.ts]: is this change intentional? i think it causes a lint error.

File: src/tests/slot-composer-test.ts [hunk 2]
```diff
@@ -132,15 +127,14 @@ recipe
...
     observer
         .newExpectations()
         .expectRenderSlot('List', 'root')
         .expectRenderSlot('List', 'root')
         .expectRenderSlot('ShowProduct', 'item')
         ;
-
-    await suggestion.instantiate(arc);
-    await arc.idle;
+    await arc.instantiate(suggestion.plan);
```
Comment (mmandlis) [inline src/tests/slot-composer-test.ts]: just curious - why did you remove the `await arc.idle`?

File: src/tests/slot-composer-test.ts (hunk 1 above)
Comment (sjmiles) [reply src/tests/slot-composer-test.ts]: Darn it. If you need `this.timeout()` you have to use a plain `function()`, but otherwise lint forces `() =>` format. I've had to keep switching in order to debug. :(

File: src/tests/slot-composer-test.ts (hunk 2 above)
Comment (sjmiles) [reply src/tests/slot-composer-test.ts]: Because observer is going to wait until expectations are met (or until timeout), waiting for the arc to go idle isn't semantically different.

File: src/runtime/ui-multiplexer-particle.ts [hunk 3]
```diff
@@ -150,12 +150,12 @@ export class UiMultiplexerParticle extends UiTransformationParticle {
         // TODO(lindner): type erasure to avoid mismatch of Store vs Handle in arc.mapHandle
         // tslint:disable-next-line: no-any
         const otherHandleStore = otherHandle.storage as any;
-        otherMappedHandles.push(`use '${await arc.mapHandle(otherHandleStore)}' as v${index}`);
+        otherMappedHandles.push(`v${index}: use '${await arc.mapHandle(otherHandleStore)}'`);
```
Comment (mmandlis) [inline src/runtime/ui-multiplexer-particle.ts]: nitnit - `v` prefix is a remainder of `view`, could you rename it to `h` (for handles)?

File: src/tests/multiplexer-integration-test.ts [hunk 4]
```diff
@@ -130,4 +130,65 @@ describe('Multiplexer', () => {
 
     DriverFactory.clearRegistrationsForTesting();
   });
+
+  // TODO(sjmiles): probably should be in particles/tests/* because of Multiplexer.js
+  // TODO(sjmiles): skipped because I cannot figure out how to access the output data
```
Comment (mmandlis) [inline src/tests/multiplexer-integration-test.ts]: the log below in the test says - unexpected output data. would you like me to patch the PR and help debug?

Comment (sjmiles) [reply src/tests/multiplexer-integration-test.ts]: Yes please, I hit a bit of a wall here. Would appreciate any insights.

Comment (sjmiles) [reply src/tests/multiplexer-integration-test.ts]: Yes, I would really appreciate an assist, if you can find the cycles.

**Comment by sjmiles** [comment]
Presubmit tests performed manually.
